googlecloud-plugins/src/main/java/co/cask/hydrator/plugin/source/BigQuerySource.java

```diff
@@ -0,0 +1,221 @@
+    context.setInput(Input.of(sourceConfig.referenceName,
+                              new SourceInputFormatProvider(JsonTextBigQueryInputFormat.class, conf)));
+  }
+
+  @Override
+  public void transform(KeyValue<LongWritable, Text> input, Emitter<StructuredRecord> emitter) throws IOException {
+    StructuredRecord record = convertToStructuredRecord(input.getValue());
+    emitter.emit(record);
+  }
+
+  private StructuredRecord convertToStructuredRecord(Text jsonText) {
+    Map<String, String> map = GSON.fromJson(jsonText.toString(), MAP_STRING_STRING_TYPE);
+    Schema outputSchema = sourceConfig.getSchema();
+    StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+    for (Schema.Field field : outputSchema.getFields()) {
+      String fieldName = field.getName();
+      Schema schema = field.getSchema();
+      Schema.Type fieldType = schema.getType();
+      switch (fieldType) {
+        case INT:
+          builder.set(fieldName, Integer.parseInt(map.get(fieldName)));
+          break;
+        case BYTES:
+          builder.set(fieldName, Byte.parseByte(map.get(fieldName)));
+          break;
+        case LONG:
+          builder.set(fieldName, Long.parseLong(map.get(fieldName)));
+          break;
+        case FLOAT:
+          builder.set(fieldName, Float.parseFloat(map.get(fieldName)));
+          break;
+        case DOUBLE:
+          builder.set(fieldName, Double.parseDouble(map.get(fieldName)));
+          break;
+        case BOOLEAN:
+          builder.set(fieldName, Boolean.parseBoolean(map.get(fieldName)));
+          break;
+        case NULL:
+          builder.set(fieldName, null);
+          break;
+        case STRING:
+          builder.set(fieldName, map.get(fieldName));
+          break;
+        default:
+          throw new UnexpectedFormatException("field type " + fieldType + " is not supported.");
+      }
+    }
+    return builder.build();
+  }
+
+  /**
+   * {@link PluginConfig} class for {@link BigQuerySource}
+   */
+  public static class BQSourceConfig extends PluginConfig {
+    private static final String PROJECTID_DESC = "The ID of the project in Google Cloud";
+    private static final String TEMP_BUCKET_DESC = "The temporary Google Cloud Storage directory to be used for " +
+                                                   "storing the intermediate results. " +
+                                                   "e.g. 'gs://bucketname/directoryname'. The directory should not " +
+                                                   "already exist. Users should manually delete this directory " +
+                                                   "afterwards to avoid any extra Google Storage charges.";
+    private static final String IMPORT_QUERY_DESC = "The SELECT query to use to import data from the specified table." +
+                                                    " For example: 'SELECT TOP(corpus, 10) as title, COUNT(*) as " +
```

Comment:
Either move space to previous line, or bring another word down. Don't start with a blank; instead, alway end a line with a blank. Be consistent!